### PR TITLE
Add metadata support for VDC networks

### PR DESCRIPTION
**Pull request by adambarreiro**
This PR aims to enhance the VCD network resources and datasources to support metadata creation, read and update.

⚠️ Depends on: https://github.com/vmware/go-vcloud-director/pull/442
Related issue: #372

**Changes:**
- Added metadata field in direct, routed and isolated network datasources/resources.
- Removed redundant code.
- Updated to the most recent Terraform SDK method signatures, with *Context methods and diag.Diagnostics.
- Updated tests accordingly to check metadata behaviour is correct.

_Note:_ v2 networks will come in another PR as I noticed too late that they would require an OpenAPI implementation in go-vcloud-director.

### CODE REVIEW

**Comment by vbauzysvmware** [comment]
PR has conflicts :)

File: vcd/resource_vcd_network_isolated.go [hunk 1]
```diff
@@ -315,11 +333,22 @@ func genericVcdNetworkIsolatedRead(d *schema.ResourceData, meta interface{}, ori
...
 		if err != nil {
-			return fmt.Errorf("[isolated network read] dhcp set %s", err)
+			return diag.Errorf("[isolated network read] dhcp set %s", err)
 		}
 	}
 	dSet(d, "description", network.OrgVDCNetwork.Description)
 
+	metadata, err := network.GetMetadata()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to find network metadata: %s", err)
```
Comment (vbauzysvmware) [inline vcd/resource_vcd_network_isolated.go:343]: Could you check all logs and error messages in this PR and add exact resource, not generic "something with network" -> better "something with isolated network". Thank you

File: vcd/resource_vcd_network_test.go [hunk 2]
```diff
@@ -958,6 +1006,10 @@ resource "vcd_network_isolated" "{{.ResourceName}}" {
     start_address = "{{.StartStaticIpAddress1}}"
     end_address   = "{{.EndStaticIpAddress1}}"
   }
+  metadata = {
+    {{.MetadataKey}} = "{{.MetadataValue}}"
+    key2 = "value2"
```
Comment (vbauzysvmware) [inline vcd/resource_vcd_network_test.go:1011]: please move `key2 = "value2"` to params/templating in this PR

**Review by vbauzysvmware — CHANGES_REQUESTED** [review]

File: vcd/resource_vcd_network_isolated.go (hunk 1 above)
Comment (adambarreiro) [reply vcd/resource_vcd_network_isolated.go]: Changed, thanks! (Except for function `createOrUpdateNetworkMetadata` which is shared among the three types of networks)

File: vcd/resource_vcd_network_test.go (hunk 2 above)
Comment (adambarreiro) [reply vcd/resource_vcd_network_test.go]: Done 👍

**Review by lvirbalas — APPROVED** [review]
Very clean :)

**Review by Didainius — APPROVED** [review]

**Review by mikeletux — APPROVED** [review]
LGTM! Approved!

**Review by vbauzysvmware — APPROVED** [review]

**Pull request merged**